apps/ensrainbow/src/commands/validate-command.ts

```diff
@@ -0,0 +1,87 @@
+import { labelHashToBytes } from "ensrainbow-sdk/label-utils";
+import { labelhash } from "viem";
+import { LABELHASH_COUNT_KEY, openDatabase, safeGet } from "../lib/database.js";
+import { byteArraysEqual } from "../utils/byte-utils.js";
+import { LogLevel, createLogger } from "../utils/logger.js";
+import { parseNonNegativeInteger } from "../utils/number-utils.js";
+
+export interface ValidateCommandOptions {
+  dataDir: string;
+  logLevel?: LogLevel;
+}
+
+export async function validateCommand(options: ValidateCommandOptions): Promise<void> {
+  const log = createLogger(options.logLevel);
+  const db = await openDatabase(options.dataDir, options.logLevel);
+
+  let totalKeys = 0;
+  let validHashes = 0;
+  let invalidHashes = 0;
+  let hashMismatches = 0;
+
+  log.info("Starting database validation...");
+
+  // Validate each key-value pair
+  for await (const [key, value] of db.iterator()) {
+    totalKeys++;
+
+    // Skip count key
```

Comment:
```suggestion
    // Skip keys not associated with rainbow records
```

Suggesting to write this in a more general way